pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -62,22 +62,18 @@ func (fs *Decomposedfs) CreateStorageSpace(ctx context.Context, req *provider.Cr
 	// "everything is a resource" this is the unique ID for the Space resource.
 	spaceID := uuid.New().String()
 	// allow sending a space id
-	if req.Opaque != nil && req.Opaque.Map != nil {
-		if e, ok := req.Opaque.Map["spaceid"]; ok && e.Decoder == "plain" {
-			spaceID = string(e.Value)
-		}
+	if reqSpaceID := utils.ReadPlainFromOpaque(req.Opaque, "spaceid"); reqSpaceID != "" {
+		spaceID = reqSpaceID
 	}
 	// allow sending a space description
-	var description string
-	if req.Opaque != nil && req.Opaque.Map != nil {
-		if e, ok := req.Opaque.Map["description"]; ok && e.Decoder == "plain" {
-			description = string(e.Value)
-		}
-	}
+	description := utils.ReadPlainFromOpaque(req.Opaque, "description")
+	// allow sending a spaceAlias
+	alias := utils.ReadPlainFromOpaque(req.Opaque, "spaceAlias")
 	// TODO enforce a uuid?
 	// TODO clarify if we want to enforce a single personal storage space or if we want to allow sending the spaceid
 	if req.Type == spaceTypePersonal {
 		spaceID = req.GetOwner().GetId().GetOpaqueId()
+		alias = spaceTypePersonal + "/" + req.GetOwner().GetUsername()
```

Comment:
This means that personal spaces will always have an `alias` and I cannot change this as a user. 

Is this intended behaviour?

Reply:
yes. We intend to resolve it in the url always as `https://cloud.owncloud.com/files/spaces/personal/einstein/relative/path/to/file.txt` appended by the ID which will result in the stable and user-friendly URL like `https://cloud.owncloud.com/files/spaces/personal/einstein/relative/path/to/file.txt?id=95b4c6f6-d8f2-4976-b0a3-1b016bdfcb7e`

Reply:
But this can be changed. We only set it during the creation. When updating the space, this can be changed also.

It does not need to be unique.

Reply:
But let us discuss the defaults.